### PR TITLE
Upgrade watchman to 4.9.0

### PR DESCRIPTION
**Pull request by thedrow**

### CODE REVIEW

**Comment by jotadrilo** [comment]
from [release notes](https://facebook.github.io/watchman/docs/release-notes.html):

> Watchman was converted from C to C++. The conversion exposed several concurrency bugs, all of which have now been fixed.

this might be useful http://jsteemann.github.io/blog/2014/10/17/using-travis-ci-for-a-c-plus-plus-11-project/

do you want to give it a try?

File: Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ WORKDIR /tmp
 
 # Install Watchman
 ENV WATCHMAN_VERSION=4.9.0
-RUN install_packages pkg-config libtool curl ca-certificates build-essential autoconf python-dev libpython-dev autotools-dev automake && \
+RUN install_packages libssl-dev pkg-config libtool curl ca-certificates build-essential autoconf python-dev libpython-dev autotools-dev automake && \
```
Comment (jotadrilo) [inline Dockerfile]: If the `dev` package is not needed at runtime, I would purge it.

**Comment by jotadrilo** [comment]
@thedrow feel free to ping me if you think these changes are enough to merge

File: Dockerfile (hunk 1 above)
Comment (thedrow) [reply Dockerfile]: I think only libssl is. I'll try later.

**Comment by jotadrilo** [comment]
If you have a chance, please try to minimize the size by leaving only runtime deps.
I would like to move this to a multistage Dockerfile in the future.

**Pull request merged**